vue-app/src/components/ButtonRow.vue

```diff
@@ -47,16 +50,23 @@
 import Vue from 'vue'
 import Component from 'vue-class-component'
 import { Prop } from 'vue-property-decorator'
+import { UPDATE_FORM_PROGRESS } from '@/store/mutation-types'
 
 @Component
 export default class ButtonRow extends Vue {
   @Prop() currentStep!: number
   @Prop() steps!: string[]
   @Prop() isStepValid!: boolean
   @Prop() callback!: () => void
+  @Prop() handleStepNav!: () => void
+  @Prop() navDisabled!: boolean
 
   handleNext(): void {
+    // Save form data (first saves when user hits Next after first step)
     this.callback()
+    // UPDATE_FORM_PROGRESS checks if furthestStep progress needs to be updated, and updates if so
+    this.$store.commit(UPDATE_FORM_PROGRESS, this.currentStep + 1)
```

Comment:
Why a separate mutation here given that the callback already updates the store?

Reply:
I believe I ran into an issue where UPDATE_FORM_PROGRESS should only be run if we're navigating forward. When I tried it the other way (all in one mutation) I recall it progressing the `furthestStep` to the next step when the user hits "Prev".
ie: 
- User completes step 1 and navigates to step 2 (furthest step is 2)
- If they click "Prev", it would update the furthestStep to 3 since we're updating the state with the presumed new `furthestStep` before we're actually navigating there.

Will look at it more of course, as there is likely a way to put it back in one mutation, but perhaps pass a boolean indicating forward or backward nav.